ydb/docs/en/core/maintenance/manual/maintenance-without-downtime.md

```diff
@@ -0,0 +1,98 @@
+    - No more than `(nto_select - 1) / 2` unavailable State Storage rings are allowed.
+- **Force** - forced mode, the failure model is ignored. Not recommended for use.
+
+### Priority {#priority}
+
+You can specify the priority of a maintenance task. A lower value means a higher priority.
+
+The actions of the task cannot be performed until all conflicting actions from tasks with a higher priority are completed. Tasks with the same priority have no advantage over each other.
+
+## Unavailable node limits {#unavailable-node-limits}
+
+In the CMS configuration, you can configure limits on the number of unavailable nodes for a database (tenant) or for the cluster as a whole. Relative and absolute limits are supported.
+
+By default, no more than 10% of unavailable nodes are allowed for each database and the cluster as a whole.
+
+## Checking algorithm {#checking-algorithm}
+
+To check if the actions of a maintenance task can be performed, the CMS sequentially goes through each action group in the task and checks the action from the group:
+- If the object of the action is a host, the CMS checks whether the action can be performed with all nodes running on the host. 
+- If the object of the action is a node, the CMS checks:
+    - Whether there is a lock on the node.
+    - Whether it's possible to lock the node according to the limits of unavailable nodes.
+    - Whether it's possible to lock all VDisks of the node according to the availability mode.
+    - Whether it's possible to lock the State Storage ring of the node according to the availability mode.
+    - Whether it's possible to lock the node according to the limit of unavailable nodes on which cluster system tablets can run.
+
+If the checks are successful, the action can be performed and a temporary locks are acquired on the checked nodes. The CMS then considers the next group of actions. Temporary locks help to understand whether the actions requested in different groups conflict with each other. Once the check is complete, the temporary locks are released.
+
+## Examples {#examples}
+
+The [ydbops](https://github.com/ydb-platform/ydbops) utility tool uses CMS for cluster maintenance without downtime. You can also use the CMS directly through the [gRPC API](https://github.com/ydb-platform/ydb/blob/main/ydb/public/api/grpc/draft/ydb_maintenance_v1.proto).
+
+### Take out a node for maintenance {#node-maintenance}
+
+{% note info "Functionality in development" %}
+
+Functionality is expected in upcoming versions of ydbops.
+
+{% endnote %}
+
+To take out a node for maintenance, you can use the command:
+```
+$ ydbops node maintenance --host <node_fqdn>
+```
+When executing this command, ydbops will acquire an exclusive lock on the node in CMS.
+
+### Rolling restart {##rolling-restart}
+
+To perform a rolling restart of the entire cluster you can use the command:
+```
+$ ydbops restart --endpoint grpc://<cluster-fqdn> --availability-mode strong
```

Comment:
Doesn't this have some additional requirements? For instance, the last time I tried (a while ago), it couldn't restart a cluster deployed with Ansible because it relied on some different systemd unit naming.

Reply:
Things have changed a bit. By default the `ydb-server-storage.service` systemd name will be used, but it is possible to specify a different systemd unit name with `--systemd-unit` flag (here is a relevant `--help` output which contains this flag: https://pastebin.com/Jqcx31Jn). 

Just for context: the fact that we have two different default unit names for deploying with Ansible and for everything else (cloud environments etc.) - is horrible and we just have to live with it for a while. 

It is probably a good idea to include the mention of `--system-unit` flag into the docs, but not in too much detail, maybe one sentence only. E.g. "If your systemd unit name is different from the default one, you may need to override it with --systemd-unit flag"

Reply:
Added aditional requirements